llama-index-integrations/llms/llama-index-llms-gemini/llama_index/llms/gemini/utils.py

```diff
@@ -105,7 +106,7 @@ def merge_neighboring_same_role_messages(
 
         # Create a new ChatMessage or similar object with merged content
         merged_message = ChatMessage(
-            role=current_message.role,
+            role=ROLES_TO_GEMINI[current_message.role],
```

Comment:
This fixes Vertex but unsure if it'll break Gemini. Do we have a unit test that will tell us?

Reply:
nope, you would have to actually call the API to confirm it works (which tbh its probably fine?)

Reply:
I don't have access to Gemini directly, only via Vertex.